R/collect.R

```diff
@@ -60,9 +84,217 @@ collect_predictions <- function(x) {
     rlang::abort(msg)
   }
 
-  collector(x, coll_col = coll_col)
+  x <- filter_predictions(x, parameters)
+
+  if (summarize) {
+    x <- average_predictions(x, parameters)
+  } else {
+    x <- collector(x, coll_col = coll_col)
+  }
+  x
+}
+
+filter_predictions <- function(x, parameters) {
+  if (is.null(parameters)) {
+    return(x)
+  }
+  metrics <- attr(x, "metrics")
+  params <- attr(x, "parameters")
```

Comment:
I'm a little bit worried about attributes being somehow dropped along the way after fitting but before `collect_predictions()` is called. I suppose it would be pretty rare

Reply:
I'll run some tests to make sure that the `dplyr` verbs don't do that. Can you think of another way to enbed them?

Reply:
All good _except_ `mutate.rset` removes them 😞

Reply:
fixed